modules/accounts.js

```diff
@@ -156,17 +158,36 @@ Accounts.prototype.setAccountAndGet = function(data, cb, tx) {
 		throw err;
 	}
 
-	library.logic.account.set(
-		address,
-		data,
-		err => {
-			if (err) {
-				return setImmediate(cb, err);
-			}
-			return library.logic.account.get({ address }, cb, tx);
-		},
-		tx
-	);
+	const task = t =>
+		new Promise((resolve, reject) => {
+			library.logic.account.set(
+				address,
+				data,
+				err => {
+					if (err) {
```

Comment:
Please follow convention for specifying error object names, to avoid any error shadowing.